### PR TITLE
Markdown link checker

### PR DESCRIPTION
**Pull request by jspaaks**
Added a GitHub action to check links within all markdown files, both outgoing links and internal links.

For the review, I have disabled the ``schedule`` part, we should enable this once this PR is approved.

At this time, there is just one broken link (celery), but that seems to be due to the site being down --the link is correct.

Refs #44

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -10,7 +10,7 @@
   - [Pull Request Guidelines](#pull-request-guidelines)
   - [Tips](#tips)
     - [Generating code from Markdown](#generating-code-from-markdown)
-    - [Generate code from Markdown and vice versa](generate-code-from-markdown-and-vice-versa)
+    - [Generate code from Markdown and vice versa](#generate-code-from-markdown-and-vice-versa)
```
Comment (sverhoeven) [inline CONTRIBUTING.md:13]: Nice evidence to have the link checker

File: .github/workflows/markdown-link-checker.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+on: [push]
+
+# on:
+#   push:
+#     branches:
+#     - master
+#   schedule:
+#   # Run everyday at 9:00 AM (See https://pubs.opengroup.org/onlinepubs/9699919799/utilities/crontab.html#tag_20_25_07)
+#   - cron: "0 9 * * *"
```
Comment (sverhoeven) [inline .github/workflows/markdown-link-checker.yml:11]: Please uncomment having the on push and on schedule both are OK to have now
```suggestion
on:
  push:
    branches:
    - master
  schedule:
  # Run everyday at 9:00 AM (See https://pubs.opengroup.org/onlinepubs/9699919799/utilities/crontab.html#tag_20_25_07)
  - cron: "0 9 * * *"
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (jspaaks) [reply CONTRIBUTING.md:13]: agree, would have been very difficult to spot otherwise

File: .github/workflows/markdown-link-checker.yml (hunk 2 above)
Comment (jspaaks) [reply .github/workflows/markdown-link-checker.yml]: done

**Comment by jspaaks** [comment]
> I corrected the celery link see 6440785, but now Zenodo seems to be unresponsive.

My experience with the broken link checker on the guide (https://github.com/NLeSC/guide/blob/632ca33de9bcc338a5c4ed3a7b3ea60081c0b968/.travis.yml#L10-L13) is that there is always going to be links that are (temporarily) down. Perhaps we can improve the GitHub Action someday by adding either a threshold like I did in the link, or retaining history-of-checking state for each link, and only return red if you checked it a couple times and all came back broken.

**Comment by sverhoeven** [comment]
Yeah, 

> > I corrected the celery link see [6440785](https://github.com/NLESC-JCER/cpp2wasm/commit/64407858dc03a310ae366fa5a3e69860b10728b5), but now Zenodo seems to be unresponsive.
> 
> My experience with the broken link checker on the guide (https://github.com/NLeSC/guide/blob/632ca33de9bcc338a5c4ed3a7b3ea60081c0b968/.travis.yml#L10-L13) is that there is always going to be links that are (temporarily) down. Perhaps we can improve the GitHub Action someday by adding either a threshold like I did in the link, or retaining history-of-checking state for each link, and only return red if you checked it a couple times and all came back broken.

Agree, but site has been down for 9 days already so using alternative url made sense in this case.

**Pull request merged**